specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json

```diff
@@ -0,0 +1,637 @@
+        ],
+        "operationId": "SapMonitors_Delete",
+        "x-ms-long-running-operation": true,
+        "summary": "Deletes a SAP monitor.",
+        "description": "Deletes a SAP monitor with the specified subscription, resource group, and monitor name.",
+        "x-ms-examples": {
+          "Deletes a SAP monitor": {
+            "$ref": "./examples/SapMonitors_Delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SapMonitorNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Deleted SapMonitor"
+          },
+          "202": {
+            "description": "Deleting SapMonitor"
+          },
+          "204": {
+            "description": "No SapMonitor to delete"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "HanaOnAzure"
+        ],
+        "operationId": "SapMonitors_Update",
+        "summary": "Patches the Tags field of a SAP monitor.",
+        "description": "Patches the Tags field of a SAP monitor for the specified subscription, resource group, and monitor name.",
+        "x-ms-examples": {
+          "Update Tags field of a SAP monitor": {
+            "$ref": "./examples/SapMonitors_PatchTags.json"
+          },
+          "Delete Tags field of a SAP monitor": {
+            "$ref": "./examples/SapMonitors_PatchTags_Delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SapMonitorNameParameter"
+          },
+          {
+            "$ref": "#/parameters/TagsParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SapMonitor"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HanaOnAzure/sapMonitors/{sapMonitorName}/providerInstances/{providerInstanceName}": {
```

Comment:
Can we call this 'providers'? This is the name of a resource type and all resources under it are implicitly referred to as Instances.

Reply:
I find "providers" not that a good choice, for two reasons:
- there is already a "providers" in the URL (before Microsoft.HanaOnAzure), so having another provider seems ambiguous;
- per our design, "providers" refer to the monitoring providers on a general level (such as saphana, prometheus etc.); however, for the sake of this API, we're dealing with instances of these providers.
That being said, "providerInstances" seems like a better choice to me.